### PR TITLE
Pull Request Etiquette and Best Practices

### PR DESCRIPTION
**Pull request by donnie-msft**
Copied directly from the Azure SDK team's repo. I read [their contributing document](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices) recently and thought it seemed appropriate that they call out PR Review etiquette. We have been wanting to improve our PR etiquette for years and I believe writing down guidelines is a step to achieving common understanding.

### CODE REVIEW

**Comment by donnie-msft** [comment]
Recreated PR as I mistakenly left the default branch name which didn't trigger CI.
Thanks @kartheekp-ms for the feedback - addressed both suggestions.
Thanks @jeffkl for the approval, please see the new PR.

File: docs/workflow.md
Comment (zivkan) [inline docs/workflow.md]: On GitHub, I disagree.

If you comment on a line of code, it is a github" conversation" where posts/comments are grouped together, and the whole thread (github calls it a conversation) can be marked as resolved. On the other hand, if you write a comment at the bottom of the "conversation" view on the PR, it's not threaded (comments not grouped together, comments are strictly chronological, and therefore when there are multiple conversaions going on, it's very difficult to track the status/outcome of each) and can't be marked as resolved.

Something you can do is use the "comment on file" feature, so you still have to pick a file to comment on, but it won't be a comment on a specific line of code.

Comment (nkolev92) [reply docs/workflow.md]: The disadvantage is that the comment will eventually disappear when the code changes. 

I don't feel too strongly either way, just providing the either perspective.

File: docs/workflow.md
Comment (nkolev92) [inline docs/workflow.md]: You need to approve the latest version, so this probably could use some workshopping to match the current process.

File: docs/workflow.md
Comment (zivkan) [reply docs/workflow.md]: FWIW, comments made at the bottom of the conversation view also don't show up here: 
image

**Comment by aortiz-msft** [comment]
Team review: We do not agree on a few of the points being suggested here. We should have a team discussion on these first.

**Comment by donnie-msft** [comment]
> Team review: We do not agree on a few of the points being suggested here. We should have a team discussion on these first.

Yes, I already had this in the team meeting agenda. If you can share your concern here, please do.

File: docs/workflow.md
Comment (donnie-msft) [reply docs/workflow.md]: Sounds like commenting on the most relevant or most prominent file without a specific line is the best compromise?

File: docs/workflow.md
Comment (donnie-msft) [reply docs/workflow.md]: I think their meaning was, if it's trivial, you can approve and mention a nit or small request.
Is the suggestion to state the flip side of that and state we must ultimately approve without **any** follow-up actions? Do we not allow trivial/nitpicks in an approval?

File: docs/workflow.md
Comment (zivkan) [reply docs/workflow.md]: That would be my vote 👍

File: docs/workflow.md
Comment (nkolev92) [reply docs/workflow.md]: I'm referring to `shared understanding with the author that the comments can be addressed or resolved prior to being merged ` 

I don't think that's referring to nits only but rather to changes that are not controversial but need addressed. 

For example, using NoAllocatorNumerator on an IList in a hot path in restore. That's something that needs addressed, but I trust that a person would do it without needing re-review. The EO meaning that latest iteration needs to be approved means that if a person addresses your feedback, they might *need* your approval again. 

It may be  as simple as we need to mention that the latest iteration of the PR needs approved.

Comment (donnie-msft) [reply docs/workflow.md]: > It may be as simple as we need to mention that the latest iteration of the PR needs approved.

It sounds like we agree. This is basically paraphrasing of my:

>  ..and state we must ultimately approve without any follow-up actions? 

It would make it less ambiguous since today we aren't very consistent with approvals. I'm personally OK with using judgement and trust, as you mention, but also sometimes it's better to just have a simple, yet hard guideline.

Comment (nkolev92) [reply docs/workflow.md]: I'd prefer something that's not strict, just make it match what we have to do since that doc was obviously written prior the EO regarding approval of latest PR iteration.

File: docs/workflow.md
Comment (donnie-msft) [inline docs/workflow.md]:
```suggestion
- Make sure your intent is clear. Is your comment about just trying to understand the code? Use softer language, such as "I'm curious ...". Do you have a knowledge that the change could introduce a regression? Make your comment emphasize that!
```

File: docs/workflow.md
Comment (donnie-msft) [inline docs/workflow.md]:
```suggestion
```

**Review by donnie-msft — COMMENTED** [review]
Proposing changes based on team meeting

File: docs/workflow.md
Comment (donnie-msft) [reply docs/workflow.md]: Mind proposing a suggestion that I can commit?

Comment (nkolev92) [reply docs/workflow.md]:
```suggestion
  - You have left comments that are uncontroversial and there is a shared understanding with the author that the comments can be addressed or resolved either within this PR or a follow up PR, without a significant change to the design or approach. If the author pushes a new change addressing those comments, you may need to re-approve as it's required that the latest iteration of a PR is approved.
 ```
 
 How's this?

**Comment by donnie-msft** [comment]
Not stale, just awaiting approvals

**Review by nkolev92 — APPROVED** [review]
I think people may want to rereview in a team meeting, but LGTM.

**Review by kartheekp-ms — APPROVED** [review]

**Review by jeffkl — APPROVED** [review]

**Comment by donnie-msft** [comment]
Merging is blocked because @aortiz-msft would like to discuss this again in the 9/12 Team Meeting.

**Review by jgonz120 — APPROVED** [review]

**Comment by donnie-msft** [comment]
Bump https://github.com/NuGet/NuGet.Client/pull/5953#issuecomment-2313743483

**Comment by donnie-msft** [comment]
Bump https://github.com/NuGet/NuGet.Client/pull/5953#issuecomment-2313743483

**Review by Nigusu-Allehu — APPROVED** [review]

**Comment by donnie-msft** [comment]
Discussed in Team Meeting 9/12. Let me know if there are suggested changes to make here.

**Comment by donnie-msft** [comment]
Bump https://github.com/NuGet/NuGet.Client/pull/5953#issuecomment-2347413443

**Review by nkolev92 — APPROVED** [review]

**Comment by nkolev92** [comment]
I'd merge this :) 5 approvals on latest (including a double one by me :D )

**Comment by donnie-msft** [comment]
> I'd merge this :) 5 approvals on latest (including a double one by me :D )

Sure! @aortiz-msft sound good?

**Pull request merged**
